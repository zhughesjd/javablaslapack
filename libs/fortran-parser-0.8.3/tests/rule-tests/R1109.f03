! Test use-stmt
!      use-stmt  is  USE [ [, module-nature ] :: ] module-name [, rename-list ]
!                or  USE [ [, module-nature ] :: ] module-name, 
!                      ONLY : [ only-list ]
!
!      module-nature  is  INTRINSIC
!                     or  NON_INTRINSIC
!
!      rename  is  local-name => use-name
!              or  OPERATOR(local-defined-operator) =>
!                    OPERATOR(use-defined-operator)
!
!      only  is  generic-spec
!            or  only-use-name
!            or  rename
!
!      only-use-name  is  use-name
!
!      local-defined-operator  is  defined-unary-op
!                              or  defined-binary-op
!
!      use-defined-operator  is  defined-unary-op
!                            or  defined-binary-op
!
! Not tested here: generic-spec, only-use-name, local-name, use-name,  
! defined-binary-op, and defined-unary-op.
!
! Note: defined-binary-op and defined-unary-op are ambiguous to the grammar 
! and are both matched as T_DEFINED_OP in the lexer.

! Test with none of the optional parts
use a

! Include optional module nature
use, intrinsic :: iso_c_binding
use, non_intrinsic :: my_mod

! Include optional rename-list
use a, b=>c, d=>e
use a, operator(.myop.)=>operator(.yourop.), integer => real, &
     b => c, operator(.myotherop.) =>operator(.yourotherop.)

! Include optional only clause
use a, only: b, c=>d

end

