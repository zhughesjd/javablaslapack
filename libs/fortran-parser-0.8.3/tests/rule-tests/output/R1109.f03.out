*******************************************
args[1]: R1109.f03
R1101:main-program__begin:
R1109:use-stmt: hasModuleNature=false hasRenameList=false hasOnly=false
R1110:module-nature:
R1109:use-stmt: hasModuleNature=true hasRenameList=false hasOnly=false
R1110:module-nature:
R1109:use-stmt: hasModuleNature=true hasRenameList=false hasOnly=false
R1111:rename-list__begin:
R1111:rename: id1=[@94,1197:1197='b',<209>,39:7] id2=[@96,1200:1200='c',<209>,39:10] defOp1=null defOp2=null
R1111:rename: id1=[@99,1203:1203='d',<209>,39:13] id2=[@101,1206:1206='e',<209>,39:16] defOp1=null defOp2=null
R1111:rename-list: count=2
R1109:use-stmt: hasModuleNature=false hasRenameList=true hasOnly=false
R1111:rename-list__begin:
R1111:rename: id1=null id2=null defOp1=[@110,1224:1229='.myop.',<192>,40:16] defOp2=[@115,1242:1249='.yourop.',<192>,40:34]
R1111:rename: id1=[@119,1253:1259='integer',<209>,40:45] id2=[@123,1264:1267='real',<209>,40:56] defOp1=null defOp2=null
R1111:rename: id1=[@133,1277:1277='b',<209>,41:5] id2=[@137,1282:1282='c',<209>,41:10] defOp1=null defOp2=null
R1111:rename: id1=null id2=null defOp1=[@142,1294:1304='.myotherop.',<192>,41:22] defOp2=[@148,1318:1330='.yourotherop.',<192>,41:46]
R1111:rename-list: count=4
R1109:use-stmt: hasModuleNature=false hasRenameList=true hasOnly=false
R1112:only-list__begin:
R1207:generic_spec: keyword=null name=[@162,1378:1378='b',<209>,44:13] type=1500
R1112:only:
R1111:rename: id1=[@165,1381:1381='c',<209>,44:16] id2=[@167,1384:1384='d',<209>,44:19] defOp1=null defOp2=null
R1112:only:
R1112:only-list: count=2
R1109:use-stmt: hasModuleNature=false hasRenameList=false hasOnly=true
R 204:specification-part: numUseStmts=6 numImportStmts=0 numDeclConstructs=0
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
